iet.so/src/action.cpp

```diff
@@ -416,34 +418,36 @@ bool iet_action::do_edp_test(map<int, uint16_t> iet_gpus_device_index) {
     int          gpuId;
     bool gpu_masking = false;    // if HIP_VISIBLE_DEVICES is set, this will be true
     int hip_num_gpu_devices;
+    unsigned int i = 0;
+    map<int, uint16_t>::iterator it;
+    uint32_t smi_num_devices;
+
     hipGetDeviceCount(&hip_num_gpu_devices);
 
     vector<IETWorker> workers(iet_gpus_device_index.size());
     for (;;) {
-        unsigned int i = 0;
```

Comment:
typically better to define variables as late as possible. Avoids unnecessary allocations in case(if later) exceptions are thrown before their usage. Though in this case not much construction or destruction takes place other than iterator, it would be better to declare them just before we need it and just before we have values needed to initialize it.

Reply:
Thought of separating declarations statements from executable statements. Will revert this change for now as there other also following the same method.